include/pybind11/class_support.h

```diff
@@ -304,6 +304,20 @@ inline void clear_instance(PyObject *self) {
     PyObject **dict_ptr = _PyObject_GetDictPtr(self);
     if (dict_ptr)
         Py_CLEAR(*dict_ptr);
+
+    if (instance->has_patients) {
```

Comment:
Since there's an `add_patient()` function, it might be nice to pair it with `clear_patients()`:
```c++
if (instance->has_patients)
    clear_patients(instance)
```